src/webapi/location.rs

```diff
@@ -1,38 +1,120 @@
 use webcore::value::Reference;
 use webcore::try_from::TryInto;
-use private::TODO;
+use webapi::dom_exception::SecurityError;
 
 /// The `Location` interface represents the location (URL) of the object it
 /// is linked to. Changes done on it are reflected on the object it relates
 /// to. Both the [Document](struct.Document.html) and [Window](struct.Window.html)
 /// interface have such a linked `Location`, accessible via [Document::location](struct.Document.html#method.location)
 /// and [Window::location](struct.Window.html#method.location) respectively.
 ///
+/// Note that all `Location` methods can return a `SecurityError` if the `Location` object's
+/// relevant `Document`'s origin is not same origin-domain with the entry settings object's origin.
+/// See: https://html.spec.whatwg.org/#dom-location-href
+///
 /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Location)
 // https://html.spec.whatwg.org/#location
 #[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
 #[reference(instance_of = "Location")]
 pub struct Location( Reference );
 
+macro_rules! jsattr {
+    [$s:ident, $attr:ident] => {{
+        js_try!(
+            return @{$s}.$attr;
+        ).unwrap()
+    }}
+}
+
 impl Location {
     /// The entire URL.
     ///
     /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Location/href)
     // https://html.spec.whatwg.org/#dom-location-href
```

Comment:
Can you change all of the specs' links to point to the IDL in the specs? (Just as the `hash` has.)